### PR TITLE
[language][diem-framework] migrate registered currency tests

### PR DESCRIPTION
**Pull request by vgao1996**
This migrates tests in `move-lang/functional-tests/diem/registered_currencies` into `diem-framework` and rewrite them as unit tests.

### CODE REVIEW

File: language/diem-framework/tests/RegisteredCurrencyTests.move [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module DiemFramework::RegisteredCurrencyTests {
```
Comment (tnowacki) [inline language/diem-framework/tests/RegisteredCurrencyTests.move:1]: `#[test_only]`?

**Review by tnowacki — COMMENTED** [review]
LGTM, will leave for @tzakian

**Review by tzakian — COMMENTED** [review]
This looks good to me other than the module not being `#[test_only]`. Once that's fixed happy to approve and get this landed!

**Comment by vgao1996** [comment]
@tzakian fixed. Thanks for reviewing!

**Review by tzakian — APPROVED** [review]

**Pull request merged**